nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/SortBlob.java

```diff
@@ -50,6 +55,32 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
         String commandId = in.getCommandId();
         Blob blob = getBlob(in.getDataAsString());
         blob = sort(blob, commandId);
+
+        File blobFile = blob.getFile();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(blobFile));
+            // Add header
+            String header = in.getHeaderAsString();
+            StringBuilder result = StringUtils.isNotEmpty(header) ? new StringBuilder(String.format("%s%n", header))
+                    : new StringBuilder();
+            String line;
+            while ((line = br.readLine()) != null) {
+                result.append(String.format("%s%n", line));
+            }
+            // Add footer
+            String footer = in.getFooterAsString();
+            if (StringUtils.isNotEmpty(footer)) {
+                result.append(footer);
+            }
+
+            blobFile.delete();
+            FileOutputStream fos = new FileOutputStream(blobFile);
+            fos.write(result.toString().getBytes());
+            fos.flush();
+        } catch (IOException e) {
+            getLog().error(e, e);
```

Comment:
Can you add an error message ?

Reply:
Done with https://jira.nuxeo.com/browse/NXP-25991